batteryInterface.h

```diff
@@ -12,20 +12,16 @@ class BatteryElements
 public:
 	static float current;

```

Comment:
Why have static in the base class?
The base class `BatteryElements` has the values and the derived class has the interfaces. Teh derived class function also accepts the actualValue and works for all range-checks. Is there a need for BatteryElements base-class still?
What if we remove it and simply pass the parameters around?
Having static variables means that their state lingers after the function, making it harder to detect side-effects.

Reply:
i will update